### PR TITLE
mypy: type more replays files

### PR DESCRIPTION
**Pull request by JoshFerge**
type a few more replays files

### CODE REVIEW

File: src/sentry/replays/endpoints/project_replay_recording_segment_details.py [hunk 1]
```diff
@@ -14,7 +15,9 @@
 
 @region_silo_endpoint
 class ProjectReplayRecordingSegmentDetailsEndpoint(ProjectEndpoint):
-    def get(self, request: Request, project, replay_id, segment_id) -> Response:
+    def get(
+        self, request: Request, project, replay_id, segment_id
+    ) -> StreamingHttpResponse | HttpResponse:
```
Comment (asottile-sentry) [inline src/sentry/replays/endpoints/project_replay_recording_segment_details.py:20]: I think the type that django usually wants here is `HttpResponseBase`

**Review by asottile-sentry — APPROVED** [review]
fixnship

File: src/sentry/replays/endpoints/project_replay_recording_segment_details.py [hunk 2]
```diff
@@ -1,9 +1,10 @@
+from __future__ import annotations
+
 from io import BytesIO
 
 import sentry_sdk
-from django.http import StreamingHttpResponse
+from django.http import HttpResponseBase
```
Comment (asottile-sentry) [inline src/sentry/replays/endpoints/project_replay_recording_segment_details.py:6]:
```suggestion
from django.http.response import HttpResponseBase
```

Comment (JoshFerge) [reply src/sentry/replays/endpoints/project_replay_recording_segment_details.py:6]: sorry tried to hack this in the github editor.. will fix

**Review by asottile-sentry — APPROVED** [review]

**Pull request merged**